apps/nrm/behaviours/application-saved.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+const saveAndExitService = require('../services/save-and-exit-service');
+
+/**
+ * Save data to database
+ *
+ * @param {object} data - request object
+ *
+ * @returns {Promise}
+ */
+const saveData = async(data) => {
+  return await saveAndExitService.write(data);
+};
+
+
+/**
+ * Get Session data
+ *
+ * @param {object} req - request object
+ *
+ * @returns {object} - the session data in the request object
+ */
+const getSessionData = (req) => {
+  const sessionData = req.sessionModel.toJSON();
+
+  return sessionData;
+};
+
+/**
+ * Save application progress
+ *
+ * Get the application data from the session then persist the data
+ * within a database
+ *
+ * @param {object} req - request object
+ *
+ * @returns {Promise} - id returned from saved application
+ */
+const saveApplication = async(req) => {
+  const sessionData = getSessionData(req);
+  return await saveData(sessionData);
+};
+
+/**
+ * Check if application has been saved successfully
+ *
+ * If the 'application-id' has been set then an application has been
+ * saved successfully
+ *
+ * @param {object} req - request object
+ *
+ * @returns {bool} - has the application been saved
+ */
+const isApplicationSaved = (req) => {
+  if (req.sessionModel.get('application-id')) {
+    return true;
+  }
+  return false;
+};
+
+module.exports = superclass => class extends superclass {
+  async getValues(req, res, next) {
+    try {
+      const applicationId = await saveApplication(req);
+      // save resulting id to session
```

Comment:
remove comment instead use:
This proves that the data has been saved to the database.